### DISCUSSION TITLE
Welcome to metrical-tracker Discussions!

### DISCUSSION THREAD

**Discussion by Zielon** [Zielon/metrical-tracker#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by emlcpfx** [comment]
Hello, this is Eric, I'm a visual effects artists in Los Angeles, and I'd love to try out this repo. I'm in the middle of downloading everything from the install script, and I have a question about the overall workflow. It sounds like you need to also have MICA installed in order to generate the identity.py files.

- Should I have that installed in a separate folder?
- If I'm doing multiple shots of the same actor (potentially hundreds,) do I need a identity.py per shot? Or is one neutral one what I'm looking for?
- I've used MediaPipe for face tracking actors in film footage, and it was very inaccurate. This looks so much better than MP, yet it's using the MP landmarks. Are you running those landmarks through your own algorithm to improve the results? (This doesn't really matter to my use case, I'm just curious about how it work.)
- I don't really understand most of the parameters in the config.py or the keyframes. Could you please explain that further? (I'm more of an artist than a programmer!)
- Does this repo export an .obj per frame? Can it export any other 3D files? Animated FBXs perhaps?

Thanks!

**Comment by Zielon** [comment]
Hello Eric,

- you can install MICA in a separate folder. It is independent of the tracker.
- you need only one image used to generate `identity.npy` and place it in your dataset.
- the landmarks are a sparse term of the optimization. Additionally, there is a dense term that is based on each pixel.
- keyframes are used during the warmup phase. It is good to select consecutive frames in a neutral pose. Like 2 or 3 frames.
- it outputs .obj per frame.

**Reply by emlcpfx** [reply]
I have it up and running, thanks for the help.

When I run tracker.py, I end up with a camera folder, but there are .jpgs in there. Is there a flag I can call, or a code change to export a camera that could be imported into Blender (or similar 3D software?)

It seems like that data is in the checkpoint files, but how do I get that data out of there and into a format readable by Blender/similar?

**Comment by DD-qi** [comment]
Dear developer, hello. Recently, I have been trying to run the program you published, and I encountered some issues while following the steps you provided. When running the demo file for MICA, I used the pre-trained model you supplied. However, when I proceeded to the step of running the tracker file (under the 'usage' module), I encountered an error message indicating that training was required (training from beginning). The specific error display was "list index out of range". Does this mean that I am required to train the MICA program?

**Reply by Zielon** [reply]
Maybe the tracker didn't find the identity.npy file in the sequence that you want to track?

**Reply by DD-qi** [reply]
Thank you for your reply. Upon your reminder, I carefully tried to reproduce the issue again and succeeded this time. Thank you for taking the time out of your busy schedule to respond to me.
